geonode/qgis_server/views.py

```diff
@@ -79,7 +79,7 @@ def download_zip(request, layername):
     return resp
 
 
-def legend(request, layername):
+def legend(request, layername, layertitle=None):
```

Comment:
Ah, I forgot to put it in urls.py. :p
So, previously the legend uses the layer filename (which I generated using tmpfile), and it looks ugly, so I hide the label. We also can use layer title on the legend, but I didn't bother searching the view that uses it :p.